pkg/operation/botanist/botanist.go

```diff
@@ -53,10 +56,22 @@ func New(o *operation.Operation) (*Botanist, error) {
 		}
 	}
 
-	if err := b.InitializeSeedClients(); err != nil {
+	if err = b.InitializeSeedClients(); err != nil {
 		return nil, err
 	}
 
+	o.Shoot.Components.DNS.ExternalProvider = b.DefaultExternalDNSProvider(b.K8sSeedClient.Client())
+	o.Shoot.Components.DNS.ExternalEntry = b.DefaultExternalDNSEntry(b.K8sSeedClient.Client())
+	o.Shoot.Components.DNS.InternalProvider = b.DefaultInternalDNSProvider(b.K8sSeedClient.Client())
+	o.Shoot.Components.DNS.InternalEntry = b.DefaultInternalDNSEntry(b.K8sSeedClient.Client())
+
+	o.Shoot.Components.DNS.Additional, err = b.AdditionalDNSProviders(context.TODO(), b.K8sGardenClient.Client(), b.K8sSeedClient.Client())
+	if err != nil {
+		return nil, err
+	}
+
+	o.Shoot.Components.Nginx.DNSEntry = b.DefaultNginxIngressDNSEntry(b.K8sSeedClient.Client())
+
```

Comment:
Can do this initialization then the operation is [created](https://github.com/gardener/gardener/blob/d70ed0f5d1a6e3523bd2cde07f5e707bec417dbf/pkg/operation/operation.go#L62)? The `Components` should be useable w/o the botanist which also prevents nil pointer dereferences.

Reply:
The seed client is not initialized before, so it's not possible without moving the client initialization as well.